File: Stack/ReverseMenu.java
import java.util.Scanner;

class ReverseMenu {
    public static void main(String[] args) {
        char stack[];
        int top=-1,c;
        String str;
        Scanner sc = new Scanner(System.in);
        while (true) {
            System.out.println("\nchoose the options: \n1. try reverse \n2. exit");
            c = sc.nextInt();
            sc.nextLine();
            switch (c) {
                case 1:
                    System.out.print("enter string = ");
                    str = sc.nextLine();

                    // created stack[] array with the size of string
                    stack = new char[str.length()];
                    
                    // inserting string elements into the stack
                    for (int i=0;i<str.length();i++) {
                        top++;
                        stack[top] = str.charAt(i);
                    }

                    // for printing output
                    System.out.print("reverse string = ");
                    for (int i=top;i>=0;i--) {
                        System.out.print(stack[i]+"");
                    }
                    break;

                case 2:
                    System.exit(0);
    
                default:
                    System.out.println("enter valid choice.");
                    break;
            }
        }
    }
}

File: Stack/ReverseMenu.c
#include<stdio.h>
#include<conio.h>
void main()
{
	int i,c,top=-1;
	char str[100],stack[100] ;
	clrscr();
  printf("\nstring reversing by using stack, array menu");
	while(1)
	{
		printf("\n1. try reverse \n2. exit \nenter your choice: ");
		scanf("%d",&c);
		switch(c)
		{
			case 1:
			printf("\nplease enter any string = ");
			scanf("%s",&str);
			for(i=0;i<strlen(str);i++)
			{
				top++;
				stack[top] = str[i];
			}

			//printing string into reverse
			printf("\nreversed string = ");
			for(i=top;i>=0;i--)
			{
				printf("%c ",stack[i]);
				top--;
			}
			break;

			case 2:
			exit(0);
			getch();
			break;

			default:
			printf("\nplease enter valid option");
			break;
		}
	}
}
